web/pdf_scripting_manager.js

```diff
@@ -203,9 +205,12 @@ class PDFScriptingManager {
   }
 
   async dispatchWillPrint() {
-    return this.#scripting?.dispatchEventInSandbox({
-      id: "doc",
-      name: "WillPrint",
+    return new Promise(resolve => {
+      this.#scripting?.dispatchEventInSandbox({
+        id: "doc",
+        name: "WillPrint",
+      });
+      this.#willPrintResolve = resolve;
     });
```

Comment:
Unfortunately this *completely breaks* printing in all documents without scripting, i.e. the majority of all PDF documents, since the Promise will never resolve in that case; did you forget to test this :-P

Additionally, if the `dispatchEventInSandbox`-call would (somehow) ever throw the new Promise would also remain pending.
Can we handle all of the above with the following implementation instead?

```suggestion
    if (!this.#scripting) {
      return undefined;
    }
    this.#willPrintCapability = new PromiseCapability();

    const result = await this.#scripting.dispatchEventInSandbox({
      id: "doc",
      name: "WillPrint",
    });
    await this.#willPrintCapability.promise;

    return result;
```